### ISSUE TITLE
Repsonse doesn't save the original batch order 

### ISSUE THREAD

**Issue by SpontaneousOverthrow** [emeraldpay/dshackle#148]
Dshackle:
```
curl --location --request POST 'http://127.0.0.1:8545' --header 'Content-Type: application/json' --data-raw '[
    {"jsonrpc":"2.0","method":"eth_chainId","params": [],"id":1},
    {"jsonrpc":"2.0","method":"eth_blockNumber","params": [],"id":2},
    {"jsonrpc":"2.0","method":"eth_chainId","params": [],"id":3}
]'

[{"jsonrpc":"2.0","id":1,"result":"0x1"},{"jsonrpc":"2.0","id":3,"result":"0x1"},{"jsonrpc":"2.0","id":2,"result":"0xd7c34b"}]
```

But when I send the same request to upstreams(erigon,geth) locally the order is persisted:

```
[{"jsonrpc":"2.0","id":1,"result":"0x1"},{"jsonrpc":"2.0","id":2,"result":"0xd7c346"},{"jsonrpc":"2.0","id":3,"result":"0x1"}]
```

This can break some popular libraries:

https://github.com/ethers-io/ethers.js/blob/master/packages/providers/src.ts/json-rpc-batch-provider.ts#L67-L77

**Comment by splix** [comment]
Well, that's expected behavior. 

That's what JSON RPC spec says https://www.jsonrpc.org/specification#batch:
> The Response objects being returned from a batch call MAY be returned in any order within the Array.

It's expected that a client uses `id` to match responses, not the order. If it doesn't do that, it's a bug on client side. Actually a very serious design flaw, not using `id` may lead to big problems.

Also, doing a batch reorder on the server side would hurt performance.

**Comment by splix** [comment]
Cannot do that on Dshackle side, should be done on ether-js side. I'm sorry, but closing the issue

**Comment by splix** [comment]
fixed in [4f1216b]
